virtcontainers/clh_test.go

```diff
@@ -0,0 +1,718 @@
+// Copyright (c) 2019 Ericsson Eurolab Deutschland G.m.b.H.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package virtcontainers
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/kata-containers/runtime/virtcontainers/device/config"
+	"github.com/kata-containers/runtime/virtcontainers/store"
+	"github.com/kata-containers/runtime/virtcontainers/types"
+	"github.com/stretchr/testify/assert"
+)
+
+//
+// Cli helper functions
+//
+
+func cliToString(args []string) string {
+	r := ""
+	for _, p := range args {
+		r += p + " "
+	}
+	return strings.TrimSpace(r)
+}
+
+func getCliOption(args []string, key string) (string, error) {
+	for i, p := range args {
+		if p == key && i < (len(args)-1) {
+			return args[i+1], nil
+		}
+	}
+	return "", errors.New("Key not found")
+}
+
+func newClhConfig() HypervisorConfig {
+	return HypervisorConfig{
+		KernelPath:        testClhKernelPath,
+		ImagePath:         testClhImagePath,
+		HypervisorPath:    testClhPath,
+		NumVCPUs:          defaultVCPUs,
+		BlockDeviceDriver: config.VirtioBlock,
+		MemorySize:        defaultMemSzMiB,
+		DefaultBridges:    defaultBridges,
+		DefaultMaxVCPUs:   MaxClhVCPUs(),
+		// Adding this here, as hypervisorconfig.valid()
+		// forcefully adds it even when 9pfs is not supported
+		Msize9p:       defaultMsize9p,
+		VirtioFSCache: virtioFsCacheAlways,
+	}
+}
+
+//
+// --cmdline <cmdline> Kernel command line
+//
+func TestClhCliKernelParameters(t *testing.T) {
+
```

Comment:
Nit: blank line at start of lots of functions.

Reply:
I think best practice to not start with newline.

Reply:
best practice is what's good for my eyes :) 
removed